### PR TITLE
chore: ignore next-env.d.ts

### PR DESCRIPTION
**Pull request by skuridin**
## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

Added `next-env.d.ts` to `.gitignore` as recommended in Next.js [documentation](https://nextjs.org/docs/basic-features/typescript).

---

## Screenshots

image

💯

### CODE REVIEW

File: .changeset/five-tigers-double.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"create-t3-app": patch
+---
+
+Add next-env.d.ts to .gitignore
```
Comment (juliusmarminge) [inline .changeset/five-tigers-double.md:5]: Should probably also delete the file https://github.com/t3-oss/create-t3-app/blob/next/cli/template/base/next-env.d.ts

**Review by nexxeln — COMMENTED** [review]
Pretty sure the docs are incorrect here. This file is required for TypeScript to work in Next.js and it should not be removed or edited. So if someone clones a project they would have to make this file on their own.

In the latest create-next-app as well this is added and committed in the initial commit.

**Comment by juliusmarminge** [comment]
> Pretty sure the docs are incorrect here. This file is required for TypeScript to work in Next.js and it should not be removed or edited. So if someone clones a project they would have to make this file on their own.
> 
> 
> 
> In the latest create-next-app as well this is added and committed in the initial commit.
> 
> 
> 
> Please make this PR to the Next.js repo and fix their docs haha.

It's generated everytime you run dev or build

**Comment by nexxeln** [comment]
> > Pretty sure the docs are incorrect here. This file is required for TypeScript to work in Next.js and it should not be removed or edited. So if someone clones a project they would have to make this file on their own.
> > In the latest create-next-app as well this is added and committed in the initial commit.
> > Please make this PR to the Next.js repo and fix their docs haha.
> 
> It's generated everytime you run dev or build

Oh I didn't know that. But then why does create-next-app put it in initial commit?

File: .changeset/five-tigers-double.md (hunk 1 above)
Comment (skuridin) [reply .changeset/five-tigers-double.md:5]: I would keep it to be consistent with `create-next-app`. It does copy this file into created app directory.

**Comment by skuridin** [comment]
> Oh I didn't know that. But then why does create-next-app put it in initial commit?

I just created new `create-next-app` and while the file is present, it is not in the "Initial commit from Create Next App" git commit.

I can only assume it's copied to the new project to avoid surprises from sudden file appearance.

**Comment by nexxeln** [comment]
> > Oh I didn't know that. But then why does create-next-app put it in initial commit?
> 
> I just created new `create-next-app` and while the file is present, it is not in the "Initial commit from Create Next App" git commit.
> 
> I can only assume it's copied to the new project to avoid surprises from sudden file appearance.

Ah okay that makes sense.

**Comment by nexxeln** [comment]
> Then we can remove this part in the docs maybe?
> 
> https://create.t3.gg/en/folder-structure#next-envdts
> 
> cc @nexxeln

It's still a file that is being generated, so I think we should keep it

File: .changeset/five-tigers-double.md (hunk 1 above)
Comment (nexxeln) [reply .changeset/five-tigers-double.md:5]: Yeah I agree we shouldn't delete

**Review by nexxeln — APPROVED** [review]
LGTM

**Pull request merged**